src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -41,9 +41,12 @@ class PeerManagerActor(
   /**
     * Maximum number of blacklisted nodes will never be larger than number of peers provided by discovery
     * Discovery provides remote nodes from all networks (ETC,ETH, Mordor etc.) only during handshake we learn that some
-    * of the remote nodes are not compatible that's why we mark them as useless (blacklist them)
+    * of the remote nodes are not compatible that's why we mark them as useless (blacklist them).
+    *
+    * The number of nodes in the current discovery is unlimited, but a guide may be the size of the routing table:
+    * one bucket for each bit in the hash of the public key, times the bucket size.
     */
-  override val maxBlacklistedNodes: Int = discoveryConfig.nodesLimit
+  override val maxBlacklistedNodes: Int = 32 * 8 * discoveryConfig.kademliaBucketSize
```

Comment:
So @ntallar this is a big change in the number of possible blacklisted nodes, I'm not sure what the implications are.

Reply:
unless we are blowing up the memory, it should be fine. All blacklists are associated time limits, so the map is cleared every now and then when time limit on blacklisted node expires.